fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/MetricUnionAvailability.java

```diff
@@ -0,0 +1,190 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.availability;
+
+import com.google.common.collect.Sets;
+import com.yahoo.bard.webservice.data.config.names.TableName;
+import com.yahoo.bard.webservice.data.metric.MetricColumn;
+import com.yahoo.bard.webservice.table.Column;
+import com.yahoo.bard.webservice.table.PhysicalTable;
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint;
+import com.yahoo.bard.webservice.util.IntervalUtils;
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList;
+
+import org.joda.time.Interval;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * An implementation of availability which which joins physical tables with the same dimension schema
+ * but non-intersecting metric schemas. Metrics on the <tt>MetricUnionAvailability</tt> are sourced from
+ * exactly one of the participating tables.
+ */
+public class MetricUnionAvailability implements Availability {
+
+    private final Set<TableName> dataSourceNames;
+    private final Set<MetricColumn> metricColumns;
+    private final Map<Availability, Set<MetricColumn>> metricAvailability;
```

Comment:
Can we give thses a name that indicates what these things _mean_ rather than what these things _are_? For example, when `metricColumns` shows up in the code of this class, it's often unclear what this collection is being used for, and a better name would help that.